orc8r/cloud/go/services/orchestrator/servicers/builder_servicer.go

```diff
@@ -191,3 +198,18 @@ func getVpnMconfig(gatewayConfig *models.MagmadGatewayConfigs) *mconfig_protos.O
 
 	return ret
 }
+
+func getStateMconfig(net configurator.Network, gwKey string) *mconfig_protos.State {
```

Comment:
new unit test here? i.e. a test that specifically runs different scenarios on `getStateMconfig`, not as part of a larger integration test on `baseOrchestratorBuilder.Build`

Reply:
getStateMconfig isn't exported - thus we should really be exercising all the paths through .Build which i beleive we are.
Re separating out test cases for detail vs passed in values through .Build there was push back on this on the original PR
https://github.com/magma/magma/pull/8482#discussion_r687209638

Reply:
strongly disagree here: targeted unit tests allow us to exercise more scenarios without a lot of overhead (all the surrounding inputs/outputs in .Build when we want to test specific scenarios in getStateMconfig)

not a make/break in this particular case, but I think we should have table-driven unit tests for private funcs (with edge condition cases) as well as typical/expected external usage unit tests for package public items.

cc @hcgatewood